### PR TITLE
WT-9838 Allow checkpoint when creating first object.

### PR DESCRIPTION
**Pull request by sueloverso**
@ddanderson I have not seen this assert fail with this change.

### CODE REVIEW

**Comment by sueloverso** [comment]
The failure is a python WT_ROLLBACK, unrelated to this change.

**Comment by sueloverso** [comment]
evergreen: retry

**Comment by sueloverso** [comment]
evergreen:retry

File: src/block_cache/block_mgr.c [hunk 1]
```diff
@@ -600,8 +600,11 @@ __bm_switch_object(WT_BM *bm, WT_SESSION_IMPL *session, uint32_t objectid)
 
     current = bm->block;
 
-    /* There should not be a checkpoint in progress. */
-    WT_ASSERT(session, !S2C(session)->txn_global.checkpoint_running);
+    /*
+     * There should not be a checkpoint in progress if we are switching objects. It is allowed if we
+     * are creating the first tiered object.
```
Comment (keitharnoldsmith) [inline src/block_cache/block_mgr.c:605]: Is this the only time where we might switch objects outside of the checkpoint lock? It looks like we could also do it (with objectid > 1) when importing a table:

https://github.com/wiredtiger/wiredtiger/blob/8be0aa4ba3878c34dbcd956e12ecc44cba094f30/src/tiered/tiered_handle.c#L765.

Comment (sueloverso) [reply src/block_cache/block_mgr.c:605]: `wt_tiered_switch` is not on the call path of `wt_btree_switch_object`. It is only called via `tiered_open` from `wt_tiered_open`.

Comment (keitharnoldsmith) [reply src/block_cache/block_mgr.c:605]: Ah.  My mistake.  I've been staring at test triage too much and my brain is a bit fried...

**Review by keitharnoldsmith — APPROVED** [review]
LGTM

**Review by ddanderson — APPROVED** [review]
LGTM

**Pull request merged**